contracts/Collector.sol

```diff
@@ -73,12 +83,23 @@ contract Collector is ICollector{
             totalShares = totalShares + _partners[i].share;
 
         require(totalShares == 100, "total shares must add up to 100%");
-
         _;
     }
 
     modifier onlyOwner(){
         require(msg.sender == owner, "can only call from owner");
         _;
     }
+
+    modifier noRemainders(){
+        uint balance = token.balanceOf(address(this));
+        require(balance < partners.length, "balance is not a remainder");
```

Comment:
I suggest a more clear message here. Something like "the balance is greater than a remainder" or "there are tokens to share" or something else.

Reply:
We are contrained by a linter error "error message for require is too long". How about "balance > remainder"?